Sources/Instrumentation/InstrumentationSystem.swift

```diff
@@ -61,24 +61,36 @@ public enum InstrumentationSystem {
         }
     }
 
-    /// Returns the globally configured `Instrument`. Defaults to a no-op `Instrument` if `boostrap` wasn't called before.
+    /// Returns the globally configured `Instrument`.
+    ///
+    /// Defaults to a no-op `Instrument` if `boostrap` wasn't called before.
     public static var instrument: Instrument {
         self.lock.withReaderLock { self._instrument }
     }
-}
 
-extension InstrumentationSystem {
     /// Get an `Instrument` instance of the given type.
     ///
     /// When using `MultiplexInstrument`, this returns the first instance of the given type stored in the `MultiplexInstrument`.
+    ///
     /// - Parameter instrumentType: The type of `Instrument` you want to retrieve an instance for.
     /// - Returns: An `Instrument` instance of the given type or `nil` if no `Instrument` of that type has been bootstrapped.
-    public static func instrument<I>(of instrumentType: I.Type) -> I? {
+    public static func instrument<I>(of instrumentType: I.Type) -> I? where I: Instrument {
+        self.__findInstrument(where: { $0 is I }) as? I
+    }
+
+}
+
+extension InstrumentationSystem {
+    /// :nodoc: INTERNAL API: Do Not Use
+    public static func __findInstrument(where predicate: (Instrument) -> Bool) -> Instrument? {
```

Comment:
Interesting. I've never seen `__` before, only `_`. Is that some kind of convention for public API that shouldn't be used outside the library?

Reply:
It could just be my convention but yea it's like "don't ever even show it in any completion and even think about using this" the :nodoc: is for jazzy to not document the API at all even tho it's public

If you remember LinuxMain those also used `public func __allTests() -> [XCTestCaseEntry] {`

Reply:
Let me confirm with some folks on guidelines...

Reply:
Got some feedback that `_` is enough to hide from autocompletion so changed that

Reply:
Okay 👍